### ISSUE TITLE
CVSplit should raise error when random_state is not used

### ISSUE THREAD

**Issue by mslapek** [skorch-dev/skorch#619]
When CVSplit is created like this:

```python
CVSplit(5, random_state=123)
```

some devs (including me! :wink:) might think, that `random_state` will cause data to be shuffled.
But it isn't.

I suggest to throw error when `random_state` is not used for given `CVSplit` (i.e. when `cv` parameter is not `float`).

It is consistent with scikit-learn:

```python
from sklearn.model_selection import KFold
a = KFold(5, shuffle=False, random_state=123)
```

gives a `FutureWarning`:

```
.../lib/python3.7/site-packages/sklearn/model_selection/_split.py:296:
FutureWarning: Setting a random_state has no effect since shuffle is False.
This will raise an error in 0.24.
You should leave random_state to its default (None), or set shuffle=True.
  FutureWarning
```

**Comment by BenjaminBossan** [comment]
To summarize: when `cv` is an integer and `random_state` is None, we should raise a `FutureWarning`, as in sklearn (not raise an error for now).

Inside of `CVSplit`, we delegate to [`check_cv`](https://github.com/scikit-learn/scikit-learn/blob/95d4f0841/sklearn/model_selection/_split.py#L1965) from sklearn. Ideally, we could pass the random state to that function and sklearn would pass it on to whatever cross validator it seems fit, giving warnings or raising errors where necessary.

Unfortunately, that is not implemented by sklearn. Therefore, to be consistent, we need to make the warning on our side. The correct place would probably be here:

https://github.com/skorch-dev/skorch/blob/7a845682be15956a132c3817f0431177a620fee5/skorch/dataset.py#L278-L283

@mslapek Would like to work on a PR to implement this?
